src/main/java/io/jenkins/plugins/jwt_auth/JwtAuthSecurityRealm.java

```diff
@@ -266,6 +275,25 @@ private Authentication getAuthFromToken(ServletRequest servletRequest) {
 					// put it in our "cache"
 					userToGroupsCache.put(username, grantedGroups);
 
+
+
+                    User user = User.getById(username, true);
+
+					if(fullNameClaim != null) {
+						String fullName = jwtClaims.getClaimValueAsString(fullNameClaim);
+						if (fullName != null) {
+							user.setFullName(fullName);
+						}
+					}
+
+					if(emailClaimName != null) {
+						String email = jwtClaims.getClaimValueAsString(emailClaimName);
+						if (email != null) {
+							user.addProperty(new Mailer.UserProperty(email));
+						}
+					}
+					user.save();
```

Comment:
Can we make this call conditional on "things having changed"?
Jenkins is rather I/O-heavy as it is, so anything we can do to reduce the amount of extra writes to disk would be A Good Thing.

i.e. Ideally, the code should detect what the existing user fullName & email is and then avoid calling user.save() if the new JWT data matched what's already there.

Reply:
@Peter-Darton-i2 Agree